### PR TITLE
Serverless Glossary

### PR DESCRIPTION
**Pull request by cswatt**
### What does this PR do?
Adds a glossary

### Preview

https://docs-staging.datadoghq.com/cswatt/serverless-glossary/serverless/glossary

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/serverless/glossary/_index.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+| Function          | In a serverless paradigm, a function is a self-contained application that runs in the cloud.                             |
+| Cloud-based development          | A workflow where developers run their code in the cloud during the development process, instead of just on their local machines. Generally, building serverless applications requires cloud-based development.                                |
+| Cold start          | The first time you invoke a particular function, it may take a longer time to run than in subsequent invocations. This phenomenon is known as a cold start, and may be caused by different factors: for instance, the cloud provider may need some time to provision underlying resources.                                |
+| Event-driven architecture          | An architectural pattern in which events drive communication between decoupled services.                                 |
+| Function-as-a-Service (FaaS)          | A subset of serverless functions. FaaS refers explicitly to event-driven paradigms.                               |
+
+## Cloud-specific concepts
```
Comment (tianchu) [inline content/en/serverless/glossary/_index.md:19]: Not too sure we want to redefine the concepts already defined by the cloud providers. Perhaps it make more sense to link the user to the cloud provider's doc?

File: content/en/serverless/glossary/_index.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+| Resource | An S3 bucket, EC2 instance, IAM user, or other entity that can be used in AWS. |
+| Resource property| When including a resource in an AWS CloudFormation stack, each resource can have one or more associated properties. |
+| Serverless Application Model (SAM)          | SAM is an Infrastructure as Code framework developed by AWS specifically focused on serverless applications.                                |
+
+
+
+### Datadog serverless for AWS Lambda concepts
+
+| Concept                         | Description                                                                                                                                                                                                          |
+|---------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (tianchu) [inline content/en/serverless/glossary/_index.md:50]: Perhaps we could also mention serverless insights?

File: content/en/serverless/glossary/_index.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+| Enhanced Lambda metrics | Enhanced Lambda metrics give you a view above and beyond the default Lambda metrics enabled with the AWS Lambda integration. These metrics are distinguished by being in the `aws.lambda.enhanced.*` namespace, and are Datadog’s best practice for setting real-time monitors on your serverless application health.|
```
Comment (tianchu) [inline content/en/serverless/glossary/_index.md:51]: Can we also add a link to the enhanced metrics doc?

File: content/en/serverless/glossary/_index.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+| Lambda library       | The Datadog Lambda Library is responsible for generating real-time enhanced Lambda metrics, submitting custom metrics, and enabling APM and Distributed Tracing for Node.js, Python, and Ruby                          |
+| Forwarder          | The Datadog Forwarder is an AWS Lambda function that ships logs, custom metrics, and traces from your environment to Datadog.                                |
+| Lambda extension          | The Datadog Lambda extension pushes real-time enhanced Lambda metrics, custom metrics, and traces from the Datadog Lambda Library to Datadog. It also forwards logs from your Lambda function to Datadog.                              |
+| Datadog Serverless CLI         | The CLI enables instrumentation by modifying existing Lambda functions' configuration. It is the quickest way to get started with Datadog serverless monitoring.                             |
```
Comment (tianchu) [inline content/en/serverless/glossary/_index.md:55]: Can probably remove `Datadog` like we don't mention `Datadog` in other rows.

Comment (tianchu) [reply content/en/serverless/glossary/_index.md:55]: It's worth mentioning the serverless macro and the serverless plugin too.

File: content/en/serverless/glossary/_index.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+
+
+### Datadog serverless for AWS Lambda concepts
+
+| Concept                         | Description                                                                                                                                                                                                          |
+|---------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Enhanced Lambda metrics | Enhanced Lambda metrics give you a view above and beyond the default Lambda metrics enabled with the AWS Lambda integration. These metrics are distinguished by being in the `aws.lambda.enhanced.*` namespace, and are Datadog’s best practice for setting real-time monitors on your serverless application health.|
+| Lambda library       | The Datadog Lambda Library is responsible for generating real-time enhanced Lambda metrics, submitting custom metrics, and enabling APM and Distributed Tracing for Node.js, Python, and Ruby                          |
```
Comment (tianchu) [inline content/en/serverless/glossary/_index.md:52]: The Datadog Lambda Library collects and submits data (enhanced lambda metrics, traces, etc.) from your Lambda function runtime either to logs (to be picked up by the Forwarder) or to the Datadog Extension.

^ not the exact language, just the idea. Better connect the dot between the Library and Forwarder/Extension.

File: content/en/serverless/glossary/_index.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+| Forwarder          | The Datadog Forwarder is an AWS Lambda function that ships logs, custom metrics, and traces from your environment to Datadog.                                |
```
Comment (tianchu) [inline content/en/serverless/glossary/_index.md:53]: The Datadog Forwarder parses and ships the serverless monitoring data from Lambda CloudWatch logs to Datadog.

File: content/en/serverless/glossary/_index.md (hunk 3 above)
Comment (tianchu) [reply content/en/serverless/glossary/_index.md:51]: Also for other items that has a doc page.

File: content/en/serverless/glossary/_index.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+| Lambda extension          | The Datadog Lambda extension pushes real-time enhanced Lambda metrics, custom metrics, and traces from the Datadog Lambda Library to Datadog. It also forwards logs from your Lambda function to Datadog.                              |
```
Comment (tianchu) [inline content/en/serverless/glossary/_index.md:54]: The Datadog Lambda Extension is a lightweight Datadog Agent that runs within the Lambda execution environment and ships serverless monitoring data to Datadog with minimal performance overhead.

File: content/en/serverless/glossary/_index.md (hunk 1 above)
Comment (cswatt) [reply content/en/serverless/glossary/_index.md:19]: I think it's helpful to consolidate here for a more full picture of what everything is — so people won't have to switch back and forth between our docs and AWS docs more than they need to

**Review by tianchu — APPROVED** [review]

File: content/en/serverless/glossary/_index.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+| [Enhanced Lambda metrics][3] | Enhanced Lambda metrics give you a view above and beyond the default Lambda metrics enabled with the AWS Lambda integration. These metrics are distinguished by being in the `aws.lambda.enhanced.*` namespace, and are Datadog’s best practice for setting real-time monitors on your serverless application health.|
+| [Lambda library][4]       | The Datadog Lambda library collects data (enhanced Lambda metrics, traces, etc.) from your Lambda function runtime. The Lambda library then submits this data either to logs (for the [Forwarder][5] to pick up) or to the [Lambda extension][6].                         |
+| [Forwarder][5]          | An AWS Lambda function that parses and ships serverless monitoring data from CloudWatch logs to Datadog.                                |
+| [Lambda extension][6]          | A lightweight Datadog Agent that runs within the Lambda execution environment and ships serverless monitoring data to Datadog with miinimal performance overhead. |
```
Comment (apigirl) [inline content/en/serverless/glossary/_index.md:54]:
```suggestion
| [Lambda extension][6]          | A lightweight Datadog Agent that runs within the Lambda execution environment and ships serverless monitoring data to Datadog with minimal performance overhead. |
```

**Review by apigirl — APPROVED** [review]

**Pull request merged**